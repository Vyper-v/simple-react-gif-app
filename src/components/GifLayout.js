import PropTypes from 'prop-types';

import Gif from './Gif';
import { useFetchGifs } from '../hooks/useFechGifs';
import './GifLayout.css'

const GifLayout = ({category}) => {
  const {data:images,loading} = useFetchGifs(category);

  return (
  <div className='grid autoLayout gap-2'>
    { loading && <p className='text-center'>Loading...</p> }
    { images.map( (img) =>  <Gif key={img.id} {...img}/> ) }
  </div>
  );
};

GifLayout.propTypes = {
  category: PropTypes.string.isRequired
};

export default GifLayout;
