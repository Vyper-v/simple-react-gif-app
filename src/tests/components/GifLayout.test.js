import GifLayout from "../../components/GifLayout";
import { shallow } from "enzyme";
import { useFetchGifs } from "../../hooks/useFechGifs";
jest.mock("../../hooks/useFechGifs");

describe("GifLayout", () => {
  const category = "One Punch";

  test("should rendered", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifLayout category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrar items cuando se cargan imagenes con useFetchGifhs", () => {
    const gifs = [
      {
        id: "ABC",
        url: "https://localhost/cualquiercosa.jpg",
        title: "Cualquier cosa",
      },
      {
        id: "123",
        url: "https://localhost/cualquiercosa.jpg",
        title: "Cualquier cosa",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifLayout category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("Gif").length).toBe(gifs.length);
  });
});
