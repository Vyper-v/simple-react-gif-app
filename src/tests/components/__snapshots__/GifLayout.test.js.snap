// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifLayout debe mostrar items cuando se cargan imagenes con useFetchGifhs 1`] = `
<div
  className="grid autoLayout gap-2"
>
  <Gif
    id="ABC"
    key="ABC"
    title="Cualquier cosa"
    url="https://localhost/cualquiercosa.jpg"
  />
  <Gif
    id="123"
    key="123"
    title="Cualquier cosa"
    url="https://localhost/cualquiercosa.jpg"
  />
</div>
`;

exports[`GifLayout should rendered 1`] = `
<div
  className="grid autoLayout gap-2"
>
  <p
    className="text-center"
  >
    Loading...
  </p>
</div>
`;
